src/fides/api/service/connectors/bigquery_connector.py

```diff
@@ -142,22 +137,40 @@ def mask_data(
         privacy_request: PrivacyRequest,
         request_task: RequestTask,
         rows: List[Row],
+        input_data: Optional[Dict[str, List[Any]]] = None,
     ) -> int:
         """Execute a masking request. Returns the number of records updated or deleted"""
+
         query_config = self.query_config(node)
         update_or_delete_ct = 0
         client = self.client()
-        for row in rows:
-            update_or_delete_stmts: List[Executable] = (
-                query_config.generate_masking_stmt(
-                    node, row, policy, privacy_request, client
-                )
-            )
-            if update_or_delete_stmts:
+
+        # Check if we're using DELETE masking strategy
+        if query_config.uses_delete_masking_strategy():
+            # Use batched DELETE for better performance
+            delete_stmts = query_config.generate_delete(client, input_data or {})
+            logger.debug(f"Generated {len(delete_stmts)} DELETE statements")
+            if delete_stmts:
                 with client.connect() as connection:
-                    for update_or_delete_stmt in update_or_delete_stmts:
-                        results: LegacyCursorResult = connection.execute(
-                            update_or_delete_stmt
-                        )
-                        update_or_delete_ct = update_or_delete_ct + results.rowcount
+                    for delete_stmt in delete_stmts:
+                        delete_results = connection.execute(delete_stmt)
+                        logger.debug(f"Deleted {delete_results.rowcount} rows")
+                        update_or_delete_ct += delete_results.rowcount
+        else:
+            # For UPDATE operations, process each row individually since each row may have different masked values
+            for row in rows:
```

Comment:
This loop is still a potentially large footgun. I'd like us to add a log like:
```suggestion
            logger.debug(f"Starting to generate and execute UPDATE statements for {len(rows)} target rows...")
            for row in rows:
```

(or similar)